internal/helpers/default_domain.go

```diff
@@ -23,3 +32,21 @@ func GetHostWithDefaultDomain(host, defaultDomainName string) string {
 func GetHostLocalDomain(host string, namespace string) string {
 	return fmt.Sprintf("%s.%s.svc.cluster.local", host, namespace)
 }
+
+func GetDefaultDomainFromKymaGateway(ctx context.Context, k8sClient client.Client) (string, error) {
+	var gateway networkingv1beta1.Gateway
+	err := k8sClient.Get(ctx, types.NamespacedName{Namespace: gatewayNamespace, Name: gatewayName}, &gateway)
+	if err != nil {
+		return "", err
+	}
+
+	if len(gateway.Spec.GetServers()) != 2 {
```

Comment:
I think we are making a lot of assumptions about the structure of the gateway. Even though this resource is under our control, for some reason it can be changed from being http the first entry. I think it would be more resilient and maybe easier to understand what data we want, if we iterate the servers and filter for the https server and using the host of this.